### DISCUSSION TITLE
Help

### DISCUSSION THREAD

**Discussion by Djsco42** [GitMyCode/Terraria-FishingBot#16] [Q&A] [answered]
GitMyCode, could you walk me though the source code of your fishing bot? I am currently learning C# and wanted to pick apart your source code, however, I do not know where to start.

**Comment by GitMyCode** [comment, marked as answer]
Well honestly this is a bit vague :) i wouldn't know how to start, what to explain or how.
Like everything you want to learn in general i would suggest you to start with something you understand and know you can build. Then go with something little bit more complicated after gaining experience.

But the gist of it is Like any wpf app the starting point is `MainWindow.xaml.cs`
when you click to run the bot it will start the loop in `Bot.cs` that launch the state machine `FishingMachine.cs`
i over complicated it with a state machine but in short this take snapshot of a part of the screen and search for the pixel of the hook you selected. this search is done in `SearchWithDeltaEColorCompare.cs`
the logic is :
when i find the hook i remember exacly the x,y position in the screen where i found it. Then i will continuously search at the position x,y to check that the hook is still there. When there is a catch there will be an animation that move the hook so that when i search again for the hook at the position x,y i will not find it. This mean there is a catch so i reel. 

Good luck and have fun

**Reply by Djsco42** [reply]
Thanks for explaining it a bit more! I might ask more questions in the future though, if that is alright with you.
